src/NuGet.Core/NuGet.Packaging/Rules/UpholdBuildConventionRule.cs

```diff
@@ -29,7 +29,10 @@ public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
             var warnings = new List<PackagingLogMessage>();
             foreach (var folder in _folders)
             {
-                var files = builder.GetFiles().Where(t => t.StartsWith(folder));
+                var files = builder.GetFiles()
+                                    .Select(t => PathUtility.GetPathWithDirectorySeparator(t))
+                                    .Where(t => t.StartsWith(folder + Path.DirectorySeparatorChar));
```

Comment:
```cs 
using System.IO;
```

Reply:
Oh, I did not compile it again after your request for rebasing :)